05/app.js

```diff
@@ -8,7 +8,37 @@ const stats = {
 };
 
 /* tutaj umieść swój kod */
+const pList = document.querySelectorAll("p[data-id]");
 
+pList.forEach(function (p) {
+  p.addEventListener("click", function (e) {
+    e.preventDefault();
+
+    const elem = e.target;
+    const tagName = e.target.tagName;
+
+    if (tagName === "P") {
+      const id = elem.dataset.id;
+
+      if (typeof stats.paragraphs[id] === "undefined") {
+        stats.paragraphs[id] = 1;
+      } else {
+        stats.paragraphs[id]++;
+      }
```

Comment:
Zwróć uwagę, że ta część jest dość podobna do tej z linii 34-38. 
Może da się zrobić z tego jedną funkcje i wykorzystać w obu miejscach?

Najpierw trzeba sprawdzić co się zmienia.
W tym konkretnym przypadku mamy albo `stats.links` albo `stats.paragraphs` to będą dane do zaktualizowania oraz mamy klucz, który może być `href` lub `id`.

Więc nasza funkcja będzie trzymywać 2 parametry:
```
function changeStats(data, key) {
}
```

Spróbuj napisać resztę :)